### ISSUE TITLE
Dialog: use h2 as default header

### ISSUE THREAD

**Issue by prsdthkr** [SAP/fundamental-styles#1115]
Component: Dialog, https://sap.github.io/fundamental-styles/components/dialog.html
Version: 0.10.0

**Expected behavior**
Though there isn't a recommended value for dialog title (i.e. heading level) in the [Fiori 3 spec](https://experience.sap.com/fiori-design-web/dialog/), the dialog examples in [UI5 and its variants](https://ui5.sap.com/#/entity/sap.m.Dialog/sample/sap.m.sample.Dialog) have h2 headers. We should consider recommending and defaulting to h2 headers for Dialog titles because most pages have a level-1 heading, and the dialog HTML is appended at the end of the DOM. This will prevent an inappropriate jump in heading hierarchy on the page which is currently being flagged as an accessibility issue by automated audit tools like AMP. Also, `fd-dialog__title` should be such that any heading level may be used with the same visual styling i.e. font-size, padding, margin etc.

**Comment by stefanoScalzo** [comment]
Has been fixed
